recipes/numexpr/meta.yaml

```diff
@@ -0,0 +1,47 @@
+{% set version="2.6.1" %}
+
+package:
+    name: numexpr
+    version: {{ version }}
+
+source:
+    fn: numexpr-{{ version}}.tar.gz
+    url: https://github.com/pydata/numexpr/archive/v{{ version }}.tar.gz
+    sha256: e92c83d066fa8da63864d69b5f218287cc31437ae844db77390f2183123aab22
+
+build:
+    number: 0
+    script: python setup.py install --single-version-externally-managed --record=record.txt
+
+requirements:
+    build:
```

Comment:
I've tried to respond to your points below. Though I'm not going to have time to debate this at present. If this is so controversial, maybe we will have to save it for the next meeting.

> those flags are not needed all the time

Unfortunately, they are needed fairly often when it comes to compiling. Particularly on OS X. As stated before, I'm not interested in setting up custom flags for everything. I tried it for awhile and it was exhausting. Also it was impossible to keep up with the rate things were being added. Hence why some things ended up using `stdlibc++` thus resulting in breakage.

> the extra flags may cause unexpected issues like the one we faced with `gdal`...making debuging hard due to the unknonwn changes

While I understand there are some special like GDAL where this has had issues (though I still need to catch up on exactly what the issues were in that case), they are pretty rare and they also tended to be cases that we already had substantial issues getting built in the first place (again GDAL). I can guarantee that `numexpr` is not one of these. I built it no problem with the `toolchain` locally. In any event, the suggestion that we reproduce and maintain these flag everywhere because there are a few cases needing special attention sounds inappropriate for the problem, unstainable in the long run, and even more problematic when mixing things as we have no real guarantees.

> the `toolchain` is not versioned in any `feedstock`, but it is constantly updated breaking build reproducibility

If it is productive for you to pin it to a version, go for it. However, the changes made of late have been relatively subtle.

> even if we start versioning them we are adding an extra layer that makes it difficult to inspect and third party using our recipes are hurt b/c of that

Not sure I follow. The few lines of code involved is open, simple, and easily gotten from `conda-forge`.

> The only cases I see the where some flags are needed is to use `clang` instead of `gcc`. If we have a `gcc` recipe in the future and adopt it everywhere the `toolchain` won't be needed at all.

Not so sure about that. See this [comment]( https://github.com/conda-forge/docker-images/pull/22#issuecomment-233403402 ). It looks like we will be playing the same game with ABI compatibility. Just we will have to do it on Linux too.

Not to mention, that you are then volunteering to update these flags in 100s of places when we switch. I certainly would not want to do that.

> But that is another topic for the future.

Agreed.